Assets/Scripts/Fire.cs

```diff
@@ -0,0 +1,50 @@
+﻿    using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class Fire : MonoBehaviour
+{
+    public float scaleFire;
+    public float speedDecreacing;
+    public float minScale;
+    public int densityFire;
+    private ParticleSystem part;
+    private ParticleSystem.ShapeModule sh;
+    private ParticleSystem.EmissionModule em;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        part = this.GetComponent<ParticleSystem>();
+        sh = part.shape;
+        em = part.emission;
+        sh.scale = Vector3.ClampMagnitude(Vector3.one, scaleFire);
+        em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(scaleFire, 3) * densityFire);
+    }
+
+    // Update is called once per frame
+     void Update()
+    {
+        Collider[] woods = Physics.OverlapSphere(this.transform.position, sh.scale.x + 1);
```

Comment:
This is a tricky issue i'm not sure what to do about myself - I think checking _every frame_ for a physics overlap can be costly for performance, especially if you do a `getComponent()` for every collider. Not sure what a good solution would be though... maybe a specific interaction when a player is holding wood to put it on the fire but that would be a shame from a gameplay perspective. Much more fun if wood can roll into it :)

Not too sure what to do about this one, maybe the cost is negligible, we should test

Reply:
So I will let it like that for the prototype and search later how to handle it better.

Reply:
Also you can use 'var' instead of Collider[], Rider tells you that normally